scripts/deploy.sh

```diff
@@ -10,6 +10,7 @@ KUBEFLOW_VERSION=${KUBEFLOW_VERSION:-"master"}
 KUBEFLOW_DEPLOY=${KUBEFLOW_DEPLOY:-true}
 K8S_NAMESPACE=${K8S_NAMESPACE:-"kubeflow"}
 KUBEFLOW_CLOUD=${KUBEFLOW_CLOUD:-"minikube"}
+KUBEFLOW_DOCKER_REGISTRY=${KUBEFLOW_DOCKER_REGISTRY:-""}
```

Comment:
All this environment variables are creating another layer of indirection and config. Basically we are creating a giant params file.

I don't think we want to keep going down that road. In fact I have a PR removing a bunch of the variables we added to start configuring GCP options.

Reply:
Please let me know which PR it is. Thanks.

Reply:
I agree. I think I have to hold on my PR when the changes are done, so I can follow the principle of community.

Reply:
I think if you pull this out into a separate script you should be ok for now.

So the instructions would be something like

1. Follow Kubeflow guide to create a ksonnet ap
2. cd to ksonnet app
3. Run script to change all the images.

Reply:
May I put this environment variable in the deploy.sh? Because I think the only difference between the separate script and the current deploy script is updating docker registry. and if we don't set `KUBEFLOW_DOCKER_REGISTRY`, there will no impact to the current deploy script.

The real challenge for me is how to name the new script. :)  deploy_with_customized_reigstry.sh?

Reply:
I want to get rid of deploy.sh; see #1308 

Users will do something like the following

kfctl.sh init ${NAME} --platform=aliyuncs
cd ${NAME}
kfctl.sh generate k8s
kfctl.sh apply k8s

Under the hood when the user calls kfctl.sh generate k8s that will check $PLATFORM and can invoke a platform dependent function. In this case we can invoke your function to customize the images.

ks init will create "env.sh" . So you can persist environment variables there during the init stage.
https://github.com/kubeflow/kubeflow/pull/1308/files#diff-8a6f2d237f052b07ca594c20830ed9d2R37

So I guess the best thing to do would be to wait for kfctl.sh to be checked in.

Reply:
Thank you. I will keep eyes on that.